viewing_party/party.py

```diff
@@ -1,23 +1,122 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    new_movie = {}
+    new_movie["title"] = title
+    new_movie["genre"] = genre
+    new_movie["rating"] = rating
+    for key in new_movie:
+        if new_movie[key] == None:
+            return new_movie[key]
+    return new_movie
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    for i in user_data["watchlist"]:
+        for key in i:
+            if key == "title" and i[key] == title:
+                user_data["watched"].append(i)
+                user_data["watchlist"].remove(i)
+    return user_data
+
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    total_rating = 0
+    if len(user_data["watched"]) == 0:
+        return total_rating
+    for i in user_data["watched"]:
+        total_rating += i["rating"]
+    rating_average = total_rating / len(user_data["watched"])
+    return rating_average
+
+def get_most_watched_genre(user_data):
 
+    if len(user_data["watched"]) == 0:
+        return None
+    else:
+        genre_list = []
+        for i in user_data["watched"]:
+            genre_list.append(i["genre"])
+        return max(genre_list, key=genre_list.count)
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+def get_unique_watched(user_data):
+    unique_list = []
+    watched_set = set()

```

Comment:
Sets are wonderful choice of data structure here! They are efficient and have all unique values, and the difference operator is very helpful. Nice!